pkg/cmd/opts/version.go

```diff
@@ -3,16 +3,98 @@ package opts
 import (
 	"strings"
 
-	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
-
-	"github.com/jenkins-x/jx/pkg/versionstream"
-
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/envctx"
+	"github.com/jenkins-x/jx/pkg/kube"
 	"github.com/jenkins-x/jx/pkg/log"
 	"github.com/jenkins-x/jx/pkg/table"
 	"github.com/jenkins-x/jx/pkg/util"
 	"github.com/jenkins-x/jx/pkg/version"
+	"github.com/jenkins-x/jx/pkg/versionstream"
+	"github.com/jenkins-x/jx/pkg/versionstream/versionstreamrepo"
 )
 
+// EnvironmentContext gets or creates a team context with the common values for working with requirements, team settings
+// and version resolvers.
+//
+// if preferRequirementsFile is true we look for the local `jx-requirements.yml` file first and then fall back to the
+// development environment settings; otherwise we try to use the requirements from the environment if present
+func (o *CommonOptions) EnvironmentContext(dir string, preferRequirementsFile bool) (*envctx.EnvironmentContext, error) {
```

Comment:
Here we seem to mix up team context and environment context - will be more understandable with a single phrase.

Reply:
I couldn't think of a better name TBH ;) - its a combination of Environment/TeamSettings/VersionResolver/RequirementsConfig - which are mostly associated with an environment. For single clusters its the dev environment which owns the team settings/requirements. For remote Staging/Production it has its own EnvironmentContext.

Any better ideas?

Reply:
one day it might be nice to combine TeamSettings/Requirements into a single object and be able to create it from an existing cluster or - if in pre-install stages inside boot commands - from disk. Though this PR is already big enough so didn't wanna do any more large refactors any time soon...

Reply:
Let me propose changes here :-)

Reply:
Ok - changes suggested - just var name really.